src/irr/asset/CXMeshFileLoader.cpp

```diff
@@ -1504,9 +1554,21 @@ bool CXMeshFileLoader::parseDataObjectMeshNormals(SContext& _ctx, SXMesh &mesh)
 
 			for (uint32_t jk=0; jk<triangles; ++jk)
 			{
-				mesh.Vertices[mesh.Indices[normalidx++]].Normal.set(normals[polygonfaces[0]]);
-				mesh.Vertices[mesh.Indices[normalidx++]].Normal.set(normals[polygonfaces[jk+1]]);
-				mesh.Vertices[mesh.Indices[normalidx++]].Normal.set(normals[polygonfaces[jk+2]]);
+				performActionBasedOnOrientationSystem
+				(
+					[&]()
+					{
+						mesh.Vertices[mesh.Indices[normalidx++]].Normal.set(normals[polygonfaces[jk + 2]]);
+						mesh.Vertices[mesh.Indices[normalidx++]].Normal.set(normals[polygonfaces[jk + 1]]);
+						mesh.Vertices[mesh.Indices[normalidx++]].Normal.set(normals[polygonfaces[0]]);
+					},
+					[&]()
+					{
+						mesh.Vertices[mesh.Indices[normalidx++]].Normal.set(normals[polygonfaces[0]]);
+						mesh.Vertices[mesh.Indices[normalidx++]].Normal.set(normals[polygonfaces[jk + 1]]);
+						mesh.Vertices[mesh.Indices[normalidx++]].Normal.set(normals[polygonfaces[jk + 2]]);
+					}
+				);
```

Comment:
where's your flip of the position and the normal around the X axis?

Also you shouldn't be inverting indices here, you'll end up premuting them twice... you'll get the same output as a result when rendering.

Reply:
oh right, I forgot about normals flip and did premuting twice actually

Reply:
being honest I don't know, I haven't done anything with multithreading before

Reply:
you're permuting the indices, and then you're permuting how normals and positions are read according to the indices... two reversals of indices == no reversal

Reply:
https://github.com/buildaworldnet/IrrlichtBAW/pull/389/commits/8b150696b2c9791cf37751d8b49c05c0b7eff34d

there is only one index mirrowing (last commit i removed double permuting), normals (added in last) and possitions are flipped as well

Reply:
where are the positions?

Reply:
here there are my boss! 
https://github.com/buildaworldnet/IrrlichtBAW/pull/389/files#diff-cb7f186202d159b4b05d2bdcc4c37d6eR989 positions
https://github.com/buildaworldnet/IrrlichtBAW/pull/389/files#diff-cb7f186202d159b4b05d2bdcc4c37d6eR989 normals
https://github.com/buildaworldnet/IrrlichtBAW/pull/389/files#diff-cb7f186202d159b4b05d2bdcc4c37d6eR989 indicies